### PR TITLE
Allow to pass flags to node.js

### LINKED ISSUE #289: [Feature]: Increase memory with a flag / Support Node Flags
Could `webpack-cli` handle the memory limit increase with a flag?

Right now we have to manually use:
```node --max_old_space_size=4096 node_modules/webpack/bin/webpack.js```

A simple flag would be very nice instead:
```webpack --increase-memory 4096```

Or multiple custom node options like:
```webpack --node.increase-memory 4096 --node.trace-deprecation --node.experimental-modules```

### LINKED ISSUE #1084: [FEATURE] Spawn a process instead of relying on main thread
**Is your feature request related to a problem? Please describe.**
On `next` branch and in general `webpack`, we're using the thread singly to run webpack. This turns out to be a problem both earlier when trying to integrate node flag support https://github.com/webpack/webpack-cli/issues/289 and in node v12 when trying to implement a wrapper to support the new modules declarations ( https://github.com/nodejs/modules/issues/389).

**Describe the solution you'd like**
Implement a child-process communication when spawning webpack in order to re-run CLI when supporting nodev12 with the experimental flag (we can check node version number for a user and append flag, then re-run a fresh process)

**Describe alternatives you've considered**
N/A

**Additional context**
This would probably take some time, but is really important to the new CLI.

### PR DESCRIPTION
**Pull request by smelukov**
**What kind of change does this PR introduce?**

feature

**Did you add tests for your changes?**

yes

**If relevant, did you update the documentation?**

About `--node-args` option

**Summary**

closes #1084
closes #289

**Does this PR introduce a breaking change?**

no

### CODE REVIEW

File: lib/bootstrap.js [hunk 1]
```diff
@@ -45,13 +47,39 @@ async function runCLI(cli, commandIsUsed) {
...
+            } else if (isNodsArg) {
+                isNodsArg = false;
+                nodeArgs.push(...value.split(' '));
+            } else {
+                cliArgs.push(value);
+            }
+        }
+
+        try {
+            const childProcess = execa('node', [...nodeArgs, cliPath, ...cliArgs], { stdio: 'inherit' });
```
Comment (ematipico) [inline lib/bootstrap.js]: At the moment `execa`is a dev dependency. You should move it inside the `dependencies`

File: lib/bootstrap.js [hunk 2]
```diff
@@ -45,13 +47,39 @@ async function runCLI(cli, commandIsUsed) {
...
 
     if (helpFlagExists) {
         cli.runHelp(process.argv);
         return;
     } else if (versionFlagExists) {
         cli.runVersion();
         return;
+    } else if (nodeArgsExists) {
+        args = cmdArgs(core, { stopAtFirstUnknown: false, partial: true });
+        const [, , ...rawArgs] = process.argv;
```
Comment (ematipico) [inline lib/bootstrap.js]: Gosh this is lovely! <3

File: lib/bootstrap.js [hunk 3]
```diff
@@ -45,13 +47,39 @@ async function runCLI(cli, commandIsUsed) {
...
+        const cliArgs = [];
+        const nodeArgs = [];
+        let isNodsArg = false;
```
Comment (ematipico) [inline lib/bootstrap.js]: Would you mind extract this logic inside a function and add some tests? 

```js
function parseArgs(rawArgs) {
  // ..
   return {
       cliArgs,
       nodeArgs
   }
}
```

File: lib/bootstrap.js (hunk 1 above)
Comment (smelukov) [reply lib/bootstrap.js]: It’s already moved ;)

File: lib/bootstrap.js (hunk 3 above)
Comment (smelukov) [reply lib/bootstrap.js]: I’ll deal with it ☺️

File: lib/bootstrap.js [hunk 4]
```diff
@@ -45,13 +47,39 @@ async function runCLI(cli, commandIsUsed) {
...
+    const nodeArgsExists = isFlagPresent(process.argv, 'node-args');
 
     if (helpFlagExists) {
         cli.runHelp(process.argv);
         return;
     } else if (versionFlagExists) {
         cli.runVersion();
         return;
+    } else if (nodeArgsExists) {
+        args = cmdArgs(core, { stopAtFirstUnknown: false, partial: true });
```
Comment (smelukov) [inline lib/bootstrap.js]: Seems like it’s useless code 😅

Comment (ematipico) [reply lib/bootstrap.js]: Didn't see it! Sorry :D

File: lib/bootstrap.js (hunk 3 above)
Comment (jamesgeorge007) [inline lib/bootstrap.js]: `isNodeArg` would sound much better.

**Comment by smelukov** [comment]
@ematipico It already works
```sh
> node cli.js --node-args "-foo=123"
node: bad option: -foo=123
```

**Comment by rishabh3112** [comment]
Thanks!

**Pull request merged**